sample_code/ep190/README.md

```diff
@@ -0,0 +1,91 @@
+# [python deque (useful interview datastructures) (intermediate)](https://youtu.be/obXt90bzgJo)
+
+Today we talk about deque and all the useful methods it has!
+
+## Interactive examples
+
+### Python
+
+```python
+import collections
+
+collections.deque
+d = collections.deque()
+d
+
+d = collections.deque((1, 2, 3))
+d
+d.pop()
+
+l = [1, 2, 3]
+l.pop()
+l
+d
+d.popleft()
+
+my_queue = collections.deque()
+my_queue.append(1)
+my_queue.append(2)
+my_queue.append(3)
+my_queue.popleft()
+
+lst = []
+lst.append(1)
+lst.append(2)
+lst.append(3)
+lst
+
+lst.pop()
+lst.pop()
+lst.pop()
+
+d
+d.appendleft(4)
+d
+
+def peek(q):
+    item = q.popleft()
+    q.appendleft(item)
+    return item
```

Comment:
I changed the `d` argument for this function to `q` in order to avoid confusion for the reader and prevent variable name shadowing.

Reply:
eh, it's not a problem in python -- there's scoping rules